src/migrations/dapps/1718910535585_create-trades-and-trade-assets-table.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { MigrationBuilder, ColumnDefinitions } from 'node-pg-migrate'
+
+export const TRADES_TABLE = 'trades'
+export const TRADE_ASSETS_TABLE = 'trade_assets'
+
+export const shorthands: ColumnDefinitions | undefined = undefined
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  pgm.createExtension('uuid-ossp', { ifNotExists: true })
+
+  pgm.createTable(TRADES_TABLE, {
+    id: {
+      type: 'uuid',
+      notNull: true,
+      primaryKey: true,
+      unique: true,
+      default: pgm.func('uuid_generate_v4()')
+    },
+    signature: {
+      type: 'text',
+      notNull: true,
+      unique: true
```

Comment:
As we're probably going to search by signature, would you mind adding an index for it?
There's the alternative as well to replace the id with a signature, as there can't be two identical signatures.

Reply:
The index is automatically created when you set the property as unique. Yes we can do it, i added the id in case we wanted to send it in other endpoints like nfts/orders so that we dont send the signature there and also if we later want the GET /trade/id endpoint